src/typecheck.rs

```diff
@@ -75,6 +86,15 @@ impl TypeCheck<operator::Expr> for lang::Expr {
                         operator: com_op.into(),
                     }))
                 }
+                lang::BinaryOp::Arithmetic(arith_op) => {
+                    Ok(operator::Expr::Arithmetic(operator::BinaryExpr::<
+                        operator::ArithmeticExpr,
+                    > {
+                        left: Box::new((*left).type_check(error_builder)?),
+                        right: Box::new((*right).type_check(error_builder)?),
+                        operator: arith_op.into(),
```

Comment:
if it's not too inconvenient, I have a mild preference for implementing `type_check` instead of `from`. I used to have `into()s` all over the place and it got super confusing & had inference problems so I switched everything to `type_check`.